crates/cheatnet/src/state.rs

```diff
@@ -291,10 +302,25 @@ impl TraceData {
             .push(new_call.clone());
 
         self.current_call_stack.push(new_call);
+        self.stack_of_resources_used_before_calls
+            .push(resources_used_before_call);
     }
 
-    pub fn exit_nested_call(&mut self) {
-        self.current_call_stack.pop();
+    pub fn exit_nested_call(&mut self, resources_used_after_call: &ExecutionResources) {
+        let last_call = self.current_call_stack.pop();
+        let resources_used_before_call = self
+            .stack_of_resources_used_before_calls
+            .pop()
+            .expect("Resources used before the call were None when exiting the call");
+
+        last_call.borrow_mut().used_execution_resources = ExecutionResources {
+            vm_resources: &resources_used_after_call.vm_resources
+                - &resources_used_before_call.vm_resources,
+            syscall_counter: subtract_syscall_counters(
+                &resources_used_after_call.syscall_counter,
+                &resources_used_before_call.syscall_counter,
+            ),
+        };
```

Comment:
perhaps a function for whole subtraction?

Reply:
I don't like it since the `vm_resources` subtraction is not implemented by us and imo it would just add another function looking like the `subtract_syscall_counters` (cannot implement a method on foreign struct).